src/patternfly/components/Masthead/masthead.scss

```
@@ -243,6 +243,10 @@ $pf-c-masthead--inset-map: build-spacer-map("none", "sm", "md", "lg", "xl", "2xl
   .pf-c-nav {
     align-self: stretch;
   }
+
+  .pf-c-button.pf-m-plain {
+    @include pf-t-dark; // for all plain buttons, especially inside other components (.pf-c-input-group), for dark mode
```

The mixin also sets `color` by default since we typically use this on blocks that inherit `color`, and `color` may not be redefined on things within the block - meaning the global var change wouldn't apply. So if the thing we apply it to uses a color other than the default, we need to pass that. https://github.com/patternfly/patternfly/blob/9ab78a81b9d6d63cd7795fa82b1ff673abba5332/src/patternfly/sass-utilities/mixins.scss#L190-L200

So here the button's default color is `Color--100` and it should be `Color--200`

```suggestion
    @include pf-t-dark(--pf-global--Color--200); // for all plain buttons, especially inside other components (.pf-c-input-group), for dark mode
```

However since the plain button defines `color`, maybe a better way to do this would be to update the extend so you can omit the `color` property from being set. That's a better way of addressing it than re-assigning the global var here in case the plain button color ever changes.